src/sinks/util/adaptive_concurrency/service.rs

```diff
@@ -275,10 +275,10 @@ mod tests {
 
         let in_flight = stats.in_flight.stats().unwrap();
         assert_eq!(in_flight.max, 1);
-        assert_eq!(in_flight.mean, 1.0);
+        approx::assert_relative_eq!(in_flight.mean, 1.0);
```

Comment:
If this comparison is failing, I think it may actually be displaying a behavior change. The math to calculate this value should be exact. Do you have a test run where this assert fails (as opposed to the one below)?